### PR TITLE
feature: add CancelOrders and CancelOrdersTo to executor

### LINKED ISSUE #482: Unstopped buy orders in backtest of the flashcrash strategy
The balance is set to 300 usdt, but the buy orders are created without stop during backtest.

[myflashcrash.log](https://github.com/c9s/bbgo/files/8254367/myflashcrash.log)
[myflashcrash.yaml.txt](https://github.com/c9s/bbgo/files/8254370/myflashcrash.yaml.txt)

### PR DESCRIPTION
**Pull request by zenixls2**
resolve #482 

add signal handling in backtest.

### CODE REVIEW

File: pkg/cmd/backtest.go [hunk 1]
```diff
@@ -324,28 +325,35 @@ var BacktestCmd = &cobra.Command{
...
-		for {
-			count := len(klineChans)
-			for _, kchanex := range klineChans {
-				kLine, more := <-kchanex.KChan
-				if more {
-					kchanex.Exchange.ConsumeKLine(kLine)
-				} else {
-					if err := kchanex.Exchange.CloseMarketData(); err != nil {
-						return err
+		shutdownCtx, cancelFunc := context.WithCancel(ctx)
```
Comment (c9s) [inline pkg/cmd/backtest.go:328]: use a separate ctx name for this? how about:

```
runCtx, cancelRun := 
```

File: pkg/cmd/backtest.go [hunk 2]
```diff
@@ -324,28 +325,35 @@ var BacktestCmd = &cobra.Command{
...
-				break
-			}
-		}
+		}()
+
+		cmdutil.WaitForSignal(shutdownCtx, syscall.SIGINT, syscall.SIGTERM)
 
 		log.Infof("shutting down trader...")
-		shutdownCtx, cancel := context.WithDeadline(ctx, time.Now().Add(10*time.Second))
+		shutdownCtx, cancelFunc = context.WithDeadline(shutdownCtx, time.Now().Add(10*time.Second))
```
Comment (c9s) [inline pkg/cmd/backtest.go:354]: shutdownCtx, cancelShutdown :=

File: pkg/strategy/xmaker/strategy.go [hunk 3]
```diff
@@ -804,7 +804,8 @@ func (s *Strategy) CrossRun(ctx context.Context, orderExecutionRouter bbgo.Order
 		defer tradeScanTicker.Stop()
 
 		defer func() {
-			if err := s.makerSession.Exchange.CancelOrders(context.Background(), s.activeMakerOrders.Orders()...); err != nil {
+			if err := orderExecutionRouter.CancelOrdersTo(context.Background(),
+				s.MakerExchange, s.activeMakerOrders.Orders()...); err != nil {
```
Comment (c9s) [inline pkg/strategy/xmaker/strategy.go:808]: this one should call GracefulCancel:

```
if err := s.activeMakerOrders.GracefulCancel(ctx, s.makerSession.Exchange); err != nil {
   /// ...
}
```

File: pkg/cmd/backtest.go (hunk 1 above)
Comment (zenixls2) [reply pkg/cmd/backtest.go]: done

File: pkg/strategy/xmaker/strategy.go (hunk 3 above)
Comment (zenixls2) [reply pkg/strategy/xmaker/strategy.go]: done

**Review by c9s — APPROVED** [review]

**Pull request merged**